### DISCUSSION TITLE
Progress tracking?

### DISCUSSION THREAD

**Discussion by informatter** [alicevision/Meshroom#2206] [Q&A]
Hi guys,

I am wondering if there is a way to programmatically detect the progress of any individual binary, for example for `aliceVision_featureExtraction`, `aliceVision_imageMatching`, `aliceVision_incrementalSfM` etc... ? The aim is to be able to detect if any step is "stuck" on a computation. Is there any log file that gets populated when there is any progress being made in any of the steps?

Any pointers would be great,

Thanks

Nicholas

**Comment by natowi** [comment]
When running the nodes from Meshroom, the progress can be viewed in two places:
![grafik](https://github.com/alicevision/Meshroom/assets/8138537/6c61fde6-de61-4da5-965a-88a20e616113)
Each node output folder contains a .log file.

The individual executables also have a log output which is updated during execution.

**Comment by informatter** [comment]
@natowi Thank you for your quick response. So is it safe to assume that if the log file for any executable is not updated after, for example 1 hour, the processing can be deemed as stuck?

**Reply by natowi** [reply]
The expected execution time is related to the hardware resources. (You´d have to run a default pipeline as benchmark and calculate the expected computation times from the results specific to your machine)

This may be interesting for you: https://github.com/alicevision/Meshroom/pull/1867 (contributions are welcome)
